browser/src/control/jsdialog/Util.ScrollableBar.ts

```diff
@@ -48,35 +48,54 @@ function showArrow(arrow: HTMLElement, show: boolean) {
 function setupResizeHandler(container: Element, scrollable: Element) {
 	const left = container.querySelector('.ui-scroll-left') as HTMLElement;
 	const right = container.querySelector('.ui-scroll-right') as HTMLElement;
+	var isRTL: boolean = document.documentElement.dir === 'rtl';
+	var timer: any; // for shift + mouse wheel up/down
+
 	const handler = function () {
 		const rootContainer = scrollable.querySelector('div');
 		if (!rootContainer) return;
 
 		if (rootContainer.scrollWidth > window.innerWidth) {
 			// we have overflowed content
-			const direction = this._RTL ? -1 : 1;
+			const direction = isRTL ? -1 : 1;
 			if (direction * scrollable.scrollLeft > 0) {
-				if (this._RTL) showArrow(right, true);
+				if (isRTL) showArrow(right, true);
 				else showArrow(left, true);
-			} else if (this._RTL) showArrow(right, false);
+			} else if (isRTL) showArrow(right, false);
 			else showArrow(left, false);
 
 			if (
 				direction * scrollable.scrollLeft <
 				rootContainer.scrollWidth - window.innerWidth - 1
 			) {
-				if (this._RTL) showArrow(left, true);
+				if (isRTL) showArrow(left, true);
 				else showArrow(right, true);
-			} else if (this._RTL) showArrow(left, false);
+			} else if (isRTL) showArrow(left, false);
 			else showArrow(right, false);
 		} else {
 			showArrow(left, false);
 			showArrow(right, false);
 		}
 	}.bind(this);
 
+	// handler for toolbar and statusbar
+	// runs if shift + mouse wheel up/down are used
+	const shiftHandler = function (e: MouseEvent) {
+		const rootContainer = scrollable.querySelector('div');
+		if (!rootContainer || !e.shiftKey) return;
+
+		clearTimeout(timer);
+		// wait until mouse wheel stops scrolling
+		timer = setTimeout(function () {
+			JSDialog.RefreshScrollables();
+		}, 350);
+	}.bind(this);
+
 	window.addEventListener('resize', handler);
 	window.addEventListener('scroll', handler);
+	document
+		.querySelector('.ui-scrollable-content')
+		.addEventListener('wheel', shiftHandler);
```

Comment:
if `setupResizeHandler` is called many times, this will add multiple 'wheel' handlers to the same objects. (every time we add to all of them). I think it should add only to the current one.

we should avoid at all cost querying `document`, please query only inside current component

Reply:
What I understand with the `query only inside current component` is that the `container: Element` parameter of the `setupResizeHandler` function. So I used it in the last commit. I hope it is OK.
Thanks.